src/Kestrel.Core/Internal/Http/Frame.cs

```diff
@@ -345,11 +345,11 @@ public void Reset()
             StatusCode = StatusCodes.Status200OK;
             ReasonPhrase = null;
 
-            RemoteIpAddress = RemoteEndPoint?.Address;
-            RemotePort = RemoteEndPoint?.Port ?? 0;
+            RemoteIpAddress = HttpConnectionFeature?.RemoteIpAddress;
+            RemotePort = HttpConnectionFeature?.RemotePort ?? 0;
```

Comment:
> Do these need to be separate things (Address and Port) rather than IPEndPoint? As its moving from a 4x property lookups to 4x interface property lookups
 
That's how it's expressed by the feature. We could have another feature but not sure if it's worth it yet.

> Also will the IHttpConnectionFeature Ip:Ports change from the initial value across resets? Cache locally then set to local values on Reset?

Good idea.